va/http.go

```diff
@@ -233,7 +236,7 @@ func (va *ValidationAuthorityImpl) newHTTPValidationTarget(
 	if !hasV6Addrs && !hasV4Addrs {
 		// If there are no v6 addrs and no v4addrs there was a bug with getAddrs or
 		// availableAddresses and we need to return an error.
-		return nil, fmt.Errorf("host %q has no IPv4 or IPv6 addresses", host)
+		return target, fmt.Errorf("host %q has no IPv4 or IPv6 addresses", host)
```

Comment:
Same here. We shouldn't be guaranteeing that the first return is usable when we've also returned an error.